packages/lib/profile/src/Profile.ts

```diff
@@ -186,21 +187,19 @@ export function isSameEnsName(
 async function createKeyPairsFromSig(
     provider: ethers.providers.JsonRpcProvider,
```

Comment:
I think it'll make more sense to pass a signer instead of a provider here. That would make the function much easier to use by components not based in the FE